extension/app/scripts.babel/pwa-check.js

```diff
@@ -14,197 +14,35 @@
-
-  return !!manifest.start_url;
-};
-
-// quoth kinlan >> re: canonical, it is always good to have them, but not needed.  This test is more of a warning than anything else to let you know you should consider it.
-var hasCanonicalUrl = _ => {
-  var link = document.querySelector('link[rel=canonical]');
-
-  return !!link;
-};
-
-var hasServiceWorkerRegistration = _ => {
-  return new Promise((resolve, reject) => {
-    navigator.serviceWorker.getRegistration().then(r => {
-      // Fail immediately for non-existent registrations.
-      if (typeof r === 'undefined') {
-        return resolve(false);
-      }
-
-      // If there's an active SW call this done.
-      if (r.active) {
-        return resolve(!!r.active);
-      }
-
-      // Give any installing SW chance to install.
-      r.installing.onstatechange = function() {
-        resolve(this.state === 'installed');
-      };
-    }).catch(_ => {
-      resolve(false);
-    });
-  });
-};
-
-var isOnHTTPS = _ => location.protocol === 'https:';
-
-function injectIntoTab(chrome, fnPair) {
-  var singleLineFn = fnPair[1].toString();
-
-  return new Promise((res, reject) => {
-    chrome.tabs.executeScript(null, {
-      code: `window.__lighthouse = window.__lighthouse || {};
-      window.__lighthouse['${fnPair[0]}'] = ${singleLineFn}`
-    }, ret => {
-      if (chrome.runtime.lastError) {
-        return reject(chrome.runtime.lastError);
-      }
-
-      res(ret);
-    });
-  });
-}
-
-function convertAuditsToPromiseStrings(audits) {
-  const auditNames = Object.keys(audits);
-
-  return auditNames.reduce((prevAuditGroup, auditName, auditGroupIndex) => {
-    // Then within each group, reduce each audit down to a Promise.
-    return prevAuditGroup + (auditGroupIndex > 0 ? ',' : '') +
-      audits[auditName].reduce((prevAudit, audit, auditIndex) => {
-        return prevAudit + (auditIndex > 0 ? ',' : '') +
-            convertAuditToPromiseString(auditName, audit);
-      }, '');
-  }, '');
-}
-
-function convertAuditToPromiseString(auditName, audit) {
-  return `Promise.all([
-      Promise.resolve("${auditName}"),
-      Promise.resolve("${audit[1]}"),
-      (${audit[0].toString()})()
-  ])`;
-}
-
-function runAudits(chrome, audits) {
-  // Reduce each group of audits.
-  const fnString = convertAuditsToPromiseStrings(audits);
-
-  // Ask the tab to run the promises, and beacon back the results.
-  chrome.tabs.executeScript(null, {
-    code: `Promise.all([${fnString}]).then(__lighthouse.postAuditResults)`
-  }, function() {
-    if (chrome.runtime.lastError) {
-      throw chrome.runtime.lastError;
-    }
-  });
-}
-
-function postAuditResults(results) {
-  chrome.runtime.sendMessage({onAuditsComplete: results});
-}
-
-var functionsToInject = [
-  ['ManifestParser', ManifestParser],
-  ['parseManifest', parseManifest],
-  ['postAuditResults', postAuditResults]
+'use strict';
+
+const ExtensionProtocol = require('../../../helpers/extension/driver.js');
```

Comment:
Dunno. I think that kind of stuff will become clearer. I wondered if we actually wanted to fold up all the node stuff into a cli folder so that the root folders are `extension`, `cli`, and other "shared" bits and pieces.